samples/bluetooth/mesh/sensor_client/src/model_handler.c

```diff
@@ -36,56 +33,112 @@ static void sensor_cli_data_cb(struct bt_mesh_sensor_cli *cli,
+	for (int chan = 0; chan < setting->type->channel_count; chan++) {
+		printk("Channel %d value: %s\n", chan,
+		       bt_mesh_sensor_ch_str(&(setting->value[chan])));
+	}
 }
 
 static const struct bt_mesh_sensor_cli_handlers bt_mesh_sensor_cli_handlers = {
 	.data = sensor_cli_data_cb,
 	.series_entry = sensor_cli_series_entry_cb,
+	.setting_status = sensor_cli_setting_status_cb,
 };
 
-static struct bt_mesh_sensor_cli sensor_cli =
-	BT_MESH_SENSOR_CLI_INIT(&bt_mesh_sensor_cli_handlers);
+static struct bt_mesh_sensor_cli sensor_cli = BT_MESH_SENSOR_CLI_INIT(&bt_mesh_sensor_cli_handlers);
 
 static struct k_work_delayable get_data_work;
 
 static void get_data(struct k_work *work)
 {
-	static bool req_temp_range;
+	static uint32_t sensor_idx;
 	int err;
 
 	/* Only one message can be published at a time. Swap sensor after each timeout. */
-	if (req_temp_range) {
+	switch ((sensor_idx++ % 3)) {
+	case (0): {
+		err = bt_mesh_sensor_cli_get(&sensor_cli, NULL, &bt_mesh_sensor_present_dev_op_temp,
+					     NULL);
+		if (err) {
+			printk("Error getting chip temperature (%d)\n", err);
+		}
+		break;
+	}
+	case (1): {
 		err = bt_mesh_sensor_cli_series_entries_get(
-			&sensor_cli, NULL,
-			&bt_mesh_sensor_rel_runtime_in_a_dev_op_temp_range, NULL, NULL,
-			NULL);
+			&sensor_cli, NULL, &bt_mesh_sensor_rel_runtime_in_a_dev_op_temp_range, NULL,
+			NULL, NULL);
 		if (err) {
-			printk("Error getting relative chip temperature data (%d)\n",
-			       err);
+			printk("Error getting relative chip temperature data (%d)\n", err);
 		}
-	} else {
-		err = bt_mesh_sensor_cli_get(
-			&sensor_cli, NULL, &bt_mesh_sensor_time_since_presence_detected,
-			NULL);
+		break;
+	}
+	case (2): {
+		err = bt_mesh_sensor_cli_get(&sensor_cli, NULL,
+					     &bt_mesh_sensor_time_since_presence_detected, NULL);
 		if (err) {
-			printk("Error getting time since presence detected (%d)\n",
-			       err);
+			printk("Error getting time since presence detected (%d)\n", err);
 		}
+		break;
+	}
 	}
+	k_work_schedule(&get_data_work, K_MSEC(GET_DATA_INTERVAL));
+}
 
-	req_temp_range = !req_temp_range;
+static uint32_t tidx;
+static const struct bt_mesh_sensor_column temp_ranges[] = {
+	{ { 0 }, { 100 } },
+	{ { 10 }, { 20 } },
+	{ { 22 }, { 30 } },
+	{ { 40 }, { 50 } },
+};
 
-	k_work_schedule(&get_data_work, K_MSEC(GET_DATA_INTERVAL));
+static void button_handler_cb(uint32_t pressed, uint32_t changed)
+{
+	if ((pressed & BIT(0))) {
+		int err;
+
+		err = bt_mesh_sensor_cli_setting_get(&sensor_cli, NULL,
+						     &bt_mesh_sensor_present_dev_op_temp,
+						     &bt_mesh_sensor_dev_op_temp_range_spec, NULL);
+		if (err) {
+			printk("Error getting range setting (%d)\n", err);
+		}
+	}
+	if ((pressed & BIT(1))) {
+		int err;
+
+		err = bt_mesh_sensor_cli_setting_set(&sensor_cli, NULL,
+						     &bt_mesh_sensor_present_dev_op_temp,
+						     &bt_mesh_sensor_dev_op_temp_range_spec,
+						     &temp_ranges[tidx++ % ARRAY_SIZE(temp_ranges)],
```

Comment:
> C:/Git_repos/ncs/nrf/samples/bluetooth/mesh/sensor_client/src/model_handler.c: In function 'button_handler_cb':
C:/Git_repos/ncs/nrf/samples/bluetooth/mesh/sensor_client/src/model_handler.c:130:12: warning: passing argument 5 of 'bt_mesh_sensor_cli_setting_set' from incompatible pointer type [-Wincompatible-pointer-types]    
  130 |            &temp_ranges[tidx++ % ARRAY_SIZE(temp_ranges)],
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |            |
      |            const struct bt_mesh_sensor_column *
In file included from C:/Git_repos/ncs/nrf/include/bluetooth/mesh/models.h:41,
                 from C:/Git_repos/ncs/nrf/samples/bluetooth/mesh/sensor_client/src/model_handler.c:8:
C:/Git_repos/ncs/nrf/include/bluetooth/mesh/sensor_cli.h:504:29: note: expected 'const struct sensor_value *' but argument is of type 'const struct bt_mesh_sensor_column *'
  504 |  const struct sensor_value *value,

Im not sure this is completly correct

Reply:
I have investigated this a bit. You could try using a multidimentional array. What about something like this?:
```
static const struct sensor_value temp_ranges[][2] = {
	{{ 0, 0 }, { 100, 0 }},
	{{ 10, 0 }, { 20, 0 }},
	{{ 22, 0 }, { 30, 0 }},
	{{ 40, 0 }, { 50, 0 }},
};

static void button_handler_cb(uint32_t pressed, uint32_t changed)
{
               ...

		err = bt_mesh_sensor_cli_setting_set(&sensor_cli, NULL,
						     &bt_mesh_sensor_present_dev_op_temp,
						     &bt_mesh_sensor_dev_op_temp_range_spec,
						     &temp_ranges[tidx++ % ARRAY_SIZE(temp_ranges)][0],
						     NULL);

               ...
}
```

Reply:
Casting the sensor column could work to, but im not sure if that is the "correct" way to handle it:
```
		err = bt_mesh_sensor_cli_setting_set(
			&sensor_cli, NULL, &bt_mesh_sensor_present_dev_op_temp,
			&bt_mesh_sensor_dev_op_temp_range_spec,
			(const struct sensor_value *)&temp_ranges[tidx++ % ARRAY_SIZE(temp_ranges)],
			NULL);
```